ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala

```diff
@@ -0,0 +1,134 @@
+package pl.touk.nussknacker.ui.process
+
+import akka.util.Timeout
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.engine.api.deployment.{ProcessActionType, ProcessState}
+import pl.touk.nussknacker.restmodel.process.ProcessIdWithName
+import pl.touk.nussknacker.restmodel.processdetails.BaseProcessDetails
+import pl.touk.nussknacker.ui.EspError.XError
+import pl.touk.nussknacker.ui.process.deployment.{Cancel, CheckStatus, Deploy}
+import pl.touk.nussknacker.ui.process.repository.ProcessRepository.ProcessNotFoundError
+import pl.touk.nussknacker.ui.process.repository.{FetchingProcessRepository, ProcessActionRepository, WriteProcessRepository}
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+import scala.concurrent.{ExecutionContext, Future}
+import pl.touk.nussknacker.engine.api.CirceUtil._
+import pl.touk.nussknacker.engine.api.deployment.ProcessActionType.ProcessActionType
+import pl.touk.nussknacker.ui.process.exception.ProcessIllegalAction
+import akka.actor.ActorRef
+import akka.pattern.ask
+import java.time
+
+/**
+  * ProcessService provides functionality for archive, unarchive, deploy, cancel process.
+  * Each action includes verification based on actual process state and checking process is subprocess / archived.
+  */
+class ProcessService(managerActor: ActorRef,
+                     duration: time.Duration,
+                     processRepository: FetchingProcessRepository[Future],
+                     processActionRepository: ProcessActionRepository,
+                     writeRepository: WriteProcessRepository) extends LazyLogging {
+
+  type EmptyResponse = XError[Unit]
+  type DeployResponse = XError[Any]
+
+  import scala.concurrent.duration._
+
+  private implicit val timeout: Timeout = Timeout(duration.toMillis millis)
+
+  /**
+    * Handling error at retrieving status from manager is created at ManagementActor
+    */
+  def getProcessState(processIdWithName: ProcessIdWithName)(implicit ec: ExecutionContext, user: LoggedUser): Future[ProcessState] = {
+    implicit val timeout: Timeout = Timeout(1 minute)
```

Comment:
Why we use fixed timeout here, instead of the one passed in constructor?

Reply:
Oh.. this is a copy from old one.. You're right.